Add wagtail aa1 to master

kleberbaum opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently there is no easy option to edit pages. #1 
- Currently the default user model is used. #2 
- Currently the default image model is used. #3 
- Currently the default image model is used. #4
- Currently there is app for global tools and snippets. #5
- Currently there is no app for navbars. #6
- Currently there is no way of storing colors in a Page. #7
- Currently there is no PostgreSQL database adapter installed. #8  
- Currently there there is no structure for atomic design. #9 
- Currently some requirements are outdated.


**What is the new behavior (if this is a feature change)?**
- Now there is a easy option to edit pages. Closes #1 
- Now a custom user model is used. Closes #2 
- Now a custom image model is used. Closes #3 
- Now a custom image model is used. Closes #4
- Now there is app for global tools and snippets. Closes #5
- Now there is a app for navbars. Closes #6
- Now there is a way of storing colors in a Page. Closes #7
- Now a PostgreSQL database adapter is installed. Closes #8  
- Now there is no structure for atomic design. Closes #9 
- Now some requirements are more up to date.


**Other information**:
- Ref: https://github.com/snek-at/tonic
- Ref: https://pypi.org/project/django-cors-headers/
- Ref: https://pypi.org/project/psycopg2/
- Ref: https://pypi.org/project/django-pattern-library/
- Ref: https://github.com/helsenorge/helsenorgelab.no

kleberbaum committed:
Add wagtail base requirements

All wagtail requirements that make sense in
a template repository has been added.

kleberbaum committed:
Update comments and formatting in settings

The comments and formatting has been updated.
Ref: https://github.com/snek-at/tonic

kleberbaum committed:
Add wagtail base settings

All wagtail base settings that make sense in
a template repository has been added.

kleberbaum committed:
Add wagtail development settings

All wagtail development settings that make sense in
a template repository has been added.

kleberbaum committed:
Add wagtail production settings

All wagtail production settings that make sense in
a template repository has been added.

kleberbaum committed:
Rework urls for wagtail

The urls have been adjusted to work with wagtail.

kleberbaum committed:
Fix typo in base settings

The quality of the code has been drastically improved.

kleberbaum committed:
Add custom user

Due to the complexity most of our projects have nowadays
we decided to add a custom user to the template.

kleberbaum committed:
Add custom document model

Due to the complexity most of our projects have nowadays
we decided to add a custom document model to the template.

kleberbaum committed:
Add custom image model

Due to the complexity most of our projects have nowadays
we decided to add a custom image model to the template.

kleberbaum committed:
Add navigation settings

Due to the complexity most of our projects have nowadays
we decided to add navigation settings to the template.

kleberbaum committed:
Add utils app

Due to the complexity most of our projects have nowadays
we decided to add a utils app with a collection of useful tools to the template.

kleberbaum committed:
Add colorfield and colorblock

Due to the complexity most of our projects have nowadays
we decided to add a custom database field for colors to the template.

kleberbaum committed:
Rework home model

The home model has been reworked.
It now contains a collection of demo Fields.

kleberbaum committed:
Add Django pattern library

A  app that helps to follow the Atomic design methodology.

kleberbaum committed:
Update django-cors-headers

django-cors-headers==3.4.0
Ref: https://pypi.org/project/django-cors-headers/

schettn reviewed (changes requested):
- We have to migrate to our new licence tag.
- Also the code doesn't comply with [Black](https://github.com/psf/black) standard.

kleberbaum committed:
Add psycopg2

PostgreSQL database adapter for the Python.
Ref: https://pypi.org/project/psycopg2/

kleberbaum committed:
Add warning to init settings

Here be dragons!!!

kleberbaum committed:
Add local settings

Add template for settings that will not be committed to git.

kleberbaum committed:
Add a directory for static src

This folder is meant to contain the source of a web app
that will be built in CICD.

kleberbaum committed:
Fix a typo in base requirements

The quality of the code has been drastically improved.

schettn committed:
Format Code

The whole project has been reformatted
using the black formatter.
Ref: https://github.com/psf/black

kleberbaum reviewed (commented):
Should be ok to merge.

pinterid commented on esite/core/apps.py:
```diff
@@ -3,7 +3,8 @@
+
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/documents/models.py:
```diff
@@ -0,0 +1,13 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast.

pinterid commented on esite/home/apps.py:
```diff
@@ -1,9 +1,9 @@
+
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/images/models.py:
```diff
@@ -0,0 +1,37 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/navigation/models.py:
```diff
@@ -0,0 +1,43 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/navigation/templatetags/navigation_tags.py:
```diff
@@ -0,0 +1,81 @@
+
+
+# Footer nav snippets
+"""
+@register.inclusion_tag('patterns/molecules/navigation/footernav.html', takes_context=True)
+def footernav(context):
+    request = context['request']
+    return {
+        'footernav': NavigationSettings.for_site(request.site).footer_navigation,
+        'request': request,
+    }
+"""
```
```suggestion
```

pinterid commented on esite/navigation/templatetags/navigation_tags.py:
```diff
@@ -0,0 +1,81 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/settings/base.py:
```diff
@@ -2,173 +2,268 @@
+ROOT_URLCONF = "esite.urls"
 
-#> Template definition
+# > Template Configuration
```
?

pinterid commented on esite/settings/base.py:
```diff
@@ -2,173 +2,268 @@
+# Wagtail forms not used so silence captcha warning
+SILENCED_SYSTEM_CHECKS = ["captcha.recaptcha_test_key_error"]
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/settings/development.py:
```diff
@@ -2,37 +2,42 @@
+BASE_URL = "http://localhost:8000"
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/settings/production.py:
```diff
@@ -19,47 +19,97 @@
+    RECAPTCHA_PRIVATE_KEY = env["RECAPTCHA_PRIVATE_KEY"]
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/urls.py:
```diff
@@ -13,12 +13,94 @@
+handler500 = "esite.utils.views.server_error"
+
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/admin.py:
```diff
@@ -0,0 +1,44 @@
+# modeladmin_register(CustomerAdminB)
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/apps.py:
```diff
@@ -0,0 +1,9 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/blocks.py:
```diff
@@ -0,0 +1,33 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/models.py:
```diff
@@ -0,0 +1,74 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/schema.py:
```diff
@@ -0,0 +1,63 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/tests.py:
```diff
@@ -0,0 +1,6 @@
+# Create your user related tests here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/views.py:
```diff
@@ -0,0 +1,36 @@
+# Create your views here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/wagtail_hooks.py:
```diff
@@ -0,0 +1,15 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/user/widgets.py:
```diff
@@ -0,0 +1,16 @@
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Change to Simon Prast

pinterid commented on esite/utils/wagtail_hooks.py:
```diff
@@ -0,0 +1,27 @@
+"""
+from wagtail.contrib.modeladmin.options import (ModelAdmin, ModelAdminGroup,
+                                                modeladmin_register)
+
+
+class CategoryModelAdmin(ModelAdmin):
+    model = Category
+    menu_icon = 'tag'
+
+
+class PersonTypeModelAdmin(ModelAdmin):
+    model = PersonType
+    menu_icon = 'tag'
+
+
+
+class TaxonomiesModelAdminGroup(ModelAdminGroup):
+    menu_label = "Taxonomies"
+    items = (
+        CategoryModelAdmin,
+        # PersonTypeModelAdmin,
+    )
+    menu_icon = 'tag'
+
+
+modeladmin_register(TaxonomiesModelAdminGroup)
+"""
```
```suggestion
```

pinterid reviewed (changes requested):
A few changes should be made

kleberbaum commented on esite/settings/base.py:
```diff
@@ -2,173 +2,268 @@
 
 import os
 
 
```
The env variable has to be assigned in the base settings.
```suggestion
env = os.environ.copy()

```

schettn committed:
Update Licenses

All license tags have been adjusted and some
missing added.

kleberbaum reviewed (commented):
Should be ok to merge now.

kleberbaum committed:
Add env variables to base settings

The env variable contains all system environment variables.

schettn committed:
Update Comments

All commented out source code
is removed now.
Also some comments have been
adjusted.

kleberbaum committed:
Add project style guide

The project style guide contains all needed pattern.
It uses the django pattern library to organize the patterns.

As a template we copied a style guide from a torchbox project.

Ref: https://pypi.org/project/django-pattern-library/
Ref: https://github.com/helsenorge/helsenorgelab.no

schettn reviewed (approved):
LGTM! :)

pinterid reviewed (approved):
Looks ok to merge.
